shuffle/move/examples/main/mod.ts

```diff
@@ -44,123 +39,93 @@ export async function printWelcome() {
-      [],
-    ),
-  );
-
-  const payload = codegen.Stdlib.encodeInitializeNftCollectionScriptFunction(
-    testNftType,
-  );
-  return await DiemHelpers.buildAndSubmitTransaction(
-    defaultUserContext.address,
-    await devapi.sequenceNumber(),
-    await defaultUserContext.readPrivateKey(),
-    payload,
+  return await invokeScriptFunction(
+    "Message::set_message",
+    [],
+    [message],
+    sender,
+    codeAddress
   );
 }
 
 // ScriptFunction example; creation of NFT.
-// See main/source/TestNFT.move
+// See main/source/nft/TestNFT.move
 export async function createTestNFTScriptFunction(
-  userContext: UserContext,
   contentUri: string,
+  sender?: UserContext,
+  codeAddress?: string,
 ) {
-  const scriptFunction: string = userContext.address +
-    "::TestNFT::create_nft";
-  const typeArguments: string[] = [];
-  const args: any[] = [contentUri];
-  return await DiemHelpers.invokeScriptFunctionForContext(
-    userContext,
-    scriptFunction,
-    typeArguments,
-    args,
+  return await invokeScriptFunction(
+    "TestNFT::create_nft",
+    [],
+    [contentUri],
+    sender,
+    codeAddress
   );
 }
 
 // ScriptFunction example; creation of NFT.
 // See main/source/TestNFT.move
 export async function transferNFTScriptFunction(
-  userContext: UserContext,
   to: string,
   creator: string,
   creationNum: StringLiteral,
+  sender?: UserContext,
+  codeAddress?: string,
 ) {
-  const scriptFunction: string = userContext.address +
-    "::NFTStandard::transfer";
-  const typeArgument = userContext.address + "::TestNFT::TestNFT";
-  const typeArguments: string[] = [typeArgument];
-  const args: any[] = [to, creator, creationNum];
-  return await DiemHelpers.invokeScriptFunctionForContext(
-    userContext,
-    scriptFunction,
-    typeArguments,
-    args,
+  codeAddress = codeAddress || defaultUserContext.address;
+
+  return await invokeScriptFunction(
+    "NFTStandard::transfer",
+    [`${codeAddress}::TestNFT::TestNFT`],
+    [to, creator, creationNum],
+    sender,
+    codeAddress
   );
 }
 
-export async function initializeNFTScriptFunction(userContext: UserContext) {
-  const scriptFunction: string = defaultUserContext.address +
-    "::NFTStandard::initialize_nft_collection";
+// Initialize NFT
+// See main/source/nft/NFTStandard.move
+// See main/source/nft/TestNFT.move
+export async function initializeNFTScriptFunction(
+  sender?: UserContext,
+  codeAddress?: string,
+) {
+  codeAddress = codeAddress || defaultUserContext.address;
+
+  return await invokeScriptFunction(
+    "NFTStandard::initialize_nft_collection",
+    [`${codeAddress}::TestNFT::TestNFT`],
+    [],
+    sender,
+    codeAddress
+  );
+}
 
-  const typeArgument = defaultUserContext.address + "::TestNFT::TestNFT";
-  const typeArguments: string[] = [typeArgument];
+export async function invokeScriptFunction(
+  funcName: string,
+  typeArgs: string[],
+  args: any[],
+  sender?: UserContext,
+  codeAddress?: string,
```

Comment:
nit: let's rename this to `moduleAddress`. That matches vocabulary elsewhere, like `ModuleId`.

Reply:
nit: Would it be simpler for us to have one `invokeScriptFunction` method in DiemHelpers that is ubiquitous, and we put these niceties in the base version? LMK what you think, I'm prioritizing API ergonomics.

DiemHelpers.invokeScriptFunctionForContext

Reply:
sounds good to me, I intent to leave that decision out of this PR, so that we can have clean delete code PR, and easier to merge